codegen/src/test/resources/software/amazon/awssdk/codegen/poet/model/basetype.java

```diff
@@ -71,12 +71,14 @@ public boolean equals(Object obj) {
     @Override
     public String toString() {
         StringBuilder sb = new StringBuilder();
-        sb.append("{");
         if (baseMember() != null) {
             sb.append("BaseMember: ").append(baseMember()).append(",");
         }
-        sb.append("}");
-        return sb.toString();
+        String str = sb.toString();
+        if (str.length() == 0) {
+            return "{}";
+        }
+        return "{" + str.substring(0, str.length() - 1) + "}";
```

Comment:
It would be more efficient to do the substring before the toString.

Reply:
I don't think there will any performance difference as both uses String constructor internally.

Using + operation for 3 strings should have pretty low effect on performance.

Reply:
`stringBuilder.setLength` shouldn't to my knowledge need to create a copy of the data like the substrings and + would need.

Reply:
👍